### ISSUE TITLE
Component is not replicating native select accessibility features

### ISSUE THREAD

**Issue by thiervoj** [TotomInc/vue3-select-component#347]
Hey !

I've been playing around with this library for a few days, it works fine, but lacks accessibility features that should be mandatory.

When comparing this component behavior to [WAI ARIA patterns'](https://www.w3.org/WAI/ARIA/apg/patterns/combobox/examples/combobox-select-only/) select behavior, we can list a few elements that are not working the same : 

1. Button has a `tabindex="-1"`, preventing users from focusing it using keyboard (#329) - Can currently be hotfixed by removing the tabindex attribute manually
2. Button should toggle the select altogether (#299) - Can currently be hotfixed by changing the button styles so that it fits its whole parent
3. When focusing the component while the options list is not shown, pressing the top or bottom arrow key should open the options list. Currently, it does nothing related to the component, and scrolls the page as if there was no focus
4. When focus is leaving the component (when clicking anywhere or using the tab key to focus something else), the options list should close, and the selected value should automatically change to the value that was focused last
5. The "Page Start" and "Page End" properties should move the focus to the first and last element in the list, respectively
6. Typing something when the element is focused should move the focus to the corresponding option (even when `is-searchable="false"`
7. And many more use cases that are described in the WAI ARIA Pattern's "Keyboard Support" section

**Comment by TotomInc** [comment]
Released in v0.14.0.
